### PR TITLE
[LLM]Support for linux package (llama, NeoX) & quantize (llama)

### PR DESCRIPTION
**Pull request by cyita**
## Description

1. Initial support of llama.cpp and redpajama.cpp on Linux by uploading .so files manually and downloading them in setup.py.
2. Support quantize(llama) on Linux.

Files need to be uploaded for Linux :
- [x] libllama_avx2.so
- [x] libllama_avx512.so
- [x] quantize-llama
- [x] libgptneox_avx2.so
- [x] libgptneox_avx512.so
- [x] quantize-gptneox

Quantize for NeoX will be added in the future.
Bloomz will be added in the future.
Py-binding will be updated in the future.

### 1. Why the change?

To support these cpp repo on Linux without exposing these repo public.

##### Build for the current platform:
`python setup.py clean --all bdist_wheel`
##### Windows:
`python setup.py clean --all bdist_wheel --win`
##### Linux：
`python setup.py clean --all bdist_wheel --linux`

#### Quantize:
```python
from bigdl.llm.ggml import quantize

quantize(input_path=r".\merged_gpt4all_llama_7b\ggml-model-f16.bin",
         output_path=r".\merged_gpt4all_llama_7b\ggml-model-q4_0.bin",
         model_family="llama",
         dtype="q4_0")
```

### 4. How to test?
- [ ] Unit test
- [x] Local test

### CODE REVIEW

File: python/llm/setup.py [hunk 1]
```diff
@@ -45,36 +45,63 @@ def get_llm_packages():
...
 ]
+lib_urls["Linux"] = [
+    "https://sourceforge.net/projects/analytics-zoo/files/bigdl-llm/libllama_avx2.so",
+    "https://sourceforge.net/projects/analytics-zoo/files/bigdl-llm/libllama_avx512.so",
+    "https://sourceforge.net/projects/analytics-zoo/files/bigdl-llm/quantize-llama",
+    "https://sourceforge.net/projects/analytics-zoo/files/bigdl-llm/libgptneox_avx2.so",
+    "https://sourceforge.net/projects/analytics-zoo/files/bigdl-llm/libgptneox_avx512.so",
+    # "https://sourceforge.net/projects/analytics-zoo/files/bigdl-llm/quantize-gptneox-avx2",
+    # "https://sourceforge.net/projects/analytics-zoo/files/bigdl-llm/quantize-gptneox-avx512",
+]
```
Comment (TheaperDeng) [inline python/llm/setup.py:64]: current build guide requires a strict order between code block 49-55 and code block 56-64. We could merge this for now but I am not sure it a good way for future developing.

File: python/llm/setup.py [hunk 2]
```diff
@@ -45,36 +45,63 @@ def get_llm_packages():
...
     libs_dir = os.path.join(llm_home, "bigdl", "llm", "libs")
     if not os.path.exists(libs_dir):
         os.makedirs(libs_dir, exist_ok=True)
     libso_file_name = url.split('/')[-1]
     libso_file = os.path.join(libs_dir, libso_file_name)
     if not os.path.exists(libso_file):
+        print(">> Downloading from ", url)
         urllib.request.urlretrieve(url, libso_file)
+        if change_permission:
+            os.chmod(libso_file, 0o775)
```
[the remark this answers is not in the document]
Comment (cyita) [reply python/llm/setup.py:77]: For Python 3 you have a prefix with 0o (zero oh). E.G, 0o666.

File: python/llm/setup.py [hunk 3]
```diff
@@ -45,36 +45,63 @@ def get_llm_packages():
     return llm_packages
```
[the remark this answers is not in the document]
Comment (cyita) [reply python/llm/setup.py:60]: Updated.

File: python/llm/setup.py (hunk 1 above)
Comment (cyita) [reply python/llm/setup.py:76]: sed -> pass platform args.

**Review by TheaperDeng — APPROVED** [review]

**Pull request merged**